docs/config.md

```diff
@@ -285,15 +285,17 @@ You can configure the set of immutable commits via
 also consider the `release@origin` bookmark immutable:
 
 ```toml
-revset-aliases."immutable_heads()" = "builtin_immutable_heads() | release@origin"
+[revset-aliases]
+"immutable_heads()" = "builtin_immutable_heads() | release@origin"
```

Comment:
Line 61 says that we prefer dotted notation but it doesn't say clearly why. I also see that our documentation is already very inconsistent about it.

One reason to prefer dotted notation is that you can then often take a single line and paste into your config. I think that's especially useful in error message and such. However, it only works to actually paste those strings as long as you consistently use dotted notation in your config file. TOML doesn't have a way to escape out of table syntax (so if there's a `[foo]` table, then you can't put a `bar.baz = 1` config after it; you have to write it `[bar]\nbaz = 1`).

So perhaps we should consistently use table notation instead. Let's hear what others think. Does anyone know what other projects do? I just checked The Cargo Book and they seem to consistently use table notation.

Reply:
I suggested the dotted notation back then as it made documenting them easier. But since we've gained a bunch of users which it may confuse, so its definitely a negative now. 

> So perhaps we should consistently use table notation instead.

Yeah, definitely in favor of that now, as our developer shortcut no longer scales.

Reply:
> Line 61 says that we prefer dotted notation

Ah yeah I see that now, but the problem is that people (e.g. me) don't sequentially read through the docs, they might have landed on a particular section via a link or a search.

Should we change the recommendation on line 61 then and change all the examples to use table notation? Happy to change this PR to do that.

Reply:
> Ah yeah I see that now, but the problem is that people (e.g. me) don't sequentially read through the docs, they might have landed on a particular section via a link or a search.

Yes, I didn't mean that you should have seen that when reading the doc :) I just meant that we apparently had a documented preference.

> Should we change the recommendation on line 61 then and change all the examples to use table notation? Happy to change this PR to do that.

Given the three thumbs up on my comment above (plus my implicit thumbs up), I think we should. Thanks!

Reply:
No worries. Done now!